lib/kokkos/core/src/desul/atomics/Generic.hpp

```diff
@@ -10,8 +10,10 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #define DESUL_ATOMICS_GENERIC_HPP_
 
 #include <type_traits>
+#if defined(__GNUC__)
```

Comment:
Should I try to get these changed merged in upstream Kokkos?

Reply:
What is currently confusing is the behavior when enabling OpenMP on MSVC for example. But also using the CLANG/LLVM variant bundled with Visual Studio is difficult since it follows the MSVC compiler conventions for flags.

Reply:
Kokkos does NOT support OpenMP with MSVC since MSVCs support is insufficient by a mile. This specific define is better protected in develop to get rid of additional warnings.

Reply:
Kokkos developers said this change is already in the Kokkos `develop` branch: https://github.com/kokkos/kokkos/blob/09a08b7bf3514114c71dde97e8f1c11050d6c733/core/src/desul/atomics/Generic.hpp#L13-L16

Reply:
Note: we protect with: 
```c++
#if defined(__GNUC__) && (!defined(__clang__))
```

Reply:
> Kokkos does NOT support OpenMP with MSVC since MSVCs support is insufficient by a mile.

If you read the entire conversation, you should have seen that I am aware of that. However, what confused me without end is that the Kokkos CMake support seems to be removing the flag that would otherwise enable OpenMP, and that it does not produce either an error in the CMake script itself or uses the value of the _OPENMP macro to state that OpenMP support is insufficient.

We are using the latter for reporting the OpenMP version with code like this:
```c++
std::string platform::openmp_standard()
{

#if !defined(_OPENMP)
  return "OpenMP not enabled";
#else

  // Supported OpenMP version corresponds to the release date of the
  // specifications as posted at https://www.openmp.org/specifications/

#if _OPENMP > 202011
  return "OpenMP newer than version 5.1";
#elif _OPENMP == 202011
  return "OpenMP 5.1";
#elif _OPENMP == 201811
  return "OpenMP 5.0";
#elif _OPENMP == 201611
  return "OpenMP 5.0 preview 1";
#elif _OPENMP == 201511
  return "OpenMP 4.5";
#elif _OPENMP == 201307
  return "OpenMP 4.0";
#elif _OPENMP == 201107
  return "OpenMP 3.1";
#elif _OPENMP == 200805
  return "OpenMP 3.0";
#elif _OPENMP == 200505
  return "OpenMP 2.5";
#elif _OPENMP == 200203
  return "OpenMP 2.0";
#else
  return "unknown OpenMP version";
#endif

#endif
}
```